intermediate_source/fx_conv_bn_fuser.py

```diff
@@ -104,7 +104,9 @@ def fuse_conv_bn_eval(conv, bn):
     module `C` such that C(x) == B(A(x)) in inference mode.
     """
     assert(not (conv.training or bn.training)), "Fusion only for eval!"
-    fused_conv = copy.deepcopy(conv)
+    fused_conv = type(conv)(conv.in_channels, conv.out_channels, conv.kernel_size)
```

Comment:
This fix seems weird? I'd guess the better is to implement a proper `__deepcopy__() for nn modules? @albanD 

This popular thread seems to validate this https://discuss.pytorch.org/t/deep-copying-pytorch-modules/13514 but idk if this is what we want people to actually do?

Reply:
> This fix seems weird? The right to do feels like its implementing a proper `__deepcopy__()` for nn modules? @albanD
> 
> This popular thread seems to validate this fix https://discuss.pytorch.org/t/deep-copying-pytorch-modules/13514 but idk if this is what we want people to actually do?

we can save and load the model, found from [2385](https://github.com/pytorch/tutorials/pull/2385) . Other than this, is there other way which i am missing that will help me make a plausible fix ?

Reply:
The problem is that Module is a complex enough class that deepcopying it is very challenging (the same way we don't recommend you serialize it as-is but only the state_dict).
deepcopy() work in most simple cases but it is expected to fail sometimes.
If you only have a regular Conv2d kernel, doing deepcopy or a new constructor is pretty much the same thing though.

Reply:
how would you want me to proceed with the PR?

Reply:
I think what @albanD is saying is that in this specific case `deepcopy`-ing a conv layer is just fine, i.e. the original code probably doesn't need to be changed.